### PR TITLE
Add DependenciesProvider

### PR DESCRIPTION
**Pull request by nataly87s**
related to #87

### CODE REVIEW

File: client/core/lib/Client.ts [hunk 1]
```diff
@@ -30,14 +30,13 @@ interface FailedRegisterPolicy {
 }
 
 export interface InitOptions {
-  prefix?: string;
   url: string;
-  cache: Storage;
+  cache: StorageController | Storage | { storage: Storage; prefix?: string };
```
Comment (eladav) [inline client/core/lib/Client.ts]: Can we create a type for `{ storage: Storage; prefix?: string }`? we can call it CacheInitOptions...

File: client/react/lib/DependenciesProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+export interface DependenciesProviderProps extends Partial<DependencyOptions> {}
+
+export const DependenciesProvider: FunctionComponent<DependenciesProviderProps> = ({ children, ...options }) => {
+  const client = useContext(DynamicoContext);
+  const overrideClient = useMemo(
+    () =>
+      client &&
+      new DynamicoClient({
+        url: client.url,
+        cache: client.cache,
```
Comment (eladav) [inline client/react/lib/DependenciesProvider.tsx]: you might want to add a special prefix here so you won't override global host component versions with ones that match only this context

**Comment by eladav** [comment]
nice change. I think that the most important thing here is changing the prefix for the overriding client's cache. I'll talk to @EladBezalel and we'll try to challenge this solution to see that everything works fine. In the meantime can you please check the the example project works the same? I think you did not break any API but just to be sure.

**Comment by nataly87s** [comment]
@eladav example project works the same

**Comment by nataly87s** [comment]
@eladav @EladBezalel what's up? any comments? requests? something?

**Comment by eladav** [comment]
@nataly87s sorry for the delay...looking at both PRs now.

**Comment by eladav** [comment]
So the reason we're having a hard time with this change is more conceptual than code-wise. The implementation looks 💯.
It looks like you want to be able to show a dynamic component from within a dynamic component. Given that you're going to use work spaces it must be quite easy to just bundle the host component along with the displayed component instead of having another client and another host-id which might make it harder for you to realize which component is shown at what configuration. 

If you worry about dependencies and bundle size it might still be easier to bundle the child component along with the host component as it already has the required dependencies.

Am I missing something? wdyt?

@EladBezalel feel free to chime in if you're available 🤔

**Comment by nataly87s** [comment]
@eladav I want to have an app with as few dependencies as possible, and a dynamic component that will inject all the other dependecies, so that if I want to update a package I won't have to publish a new version of the app, just a new version of the main component.
The app will only have native dependencies along with react and dynamico.

**Comment by nataly87s** [comment]
@eladav @EladBezalel sup?

**Pull request closed without merging**